src/main/java/uk/gov/register/store/DataAccessLayer.java

```diff
@@ -22,35 +19,38 @@
     Optional<Entry> getEntry(int entryNumber);
     Collection<Entry> getEntries(int start, int limit);
     Collection<Entry> getAllEntries();
+
     Iterator<Entry> getEntryIterator();
     Iterator<Entry> getEntryIterator(int totalEntries1, int totalEntries2);
     Iterator<Entry> getIndexEntryIterator(String indexName);
     Iterator<Entry> getIndexEntryIterator(String indexName, int totalEntries1, int totalEntries2);
     <R> R withEntryIterator(Function<EntryIterator, R> callback);
     int getTotalEntries();
     int getTotalEntries(EntryType entryType);
+    int getTotalStagedEntries(EntryType entryType);
     Optional<Instant> getLastUpdatedTime();
 
     // Item Store
     void putItem(Item item);
     Optional<Item> getItemBySha256(HashValue hash);
     Collection<Item> getAllItems();
+
     Iterator<Item> getItemIterator();
     Iterator<Item> getItemIterator(int start, int end);
     Iterator<Item> getSystemItemIterator();
     
     // Record Index
-    void updateRecordIndex(Entry entry);
     Optional<Record> getRecord(String key);
+    List<Record> getRecords(List<String> keys);
     List<Record> getRecords(int limit, int offset);
     List<Record> findMax100RecordsByKeyValue(String key, String value);
     Collection<Entry> getAllEntriesByKey(String key);
-    //int getTotalRecords();
     
     // Index
     void start(String indexName, String key, String itemHash, int startEntryNumber, int startIndexEntryNumber);
-    void end(String indexName, String entryKey, String indexKey, String itemHash, int endEntryNumber, int endIndexEntryNumber);
+    void end(String indexName, String entryKey, String indexKey, String itemHash, int endEntryNumber, int endIndexEntryNumber, int entryNumberToEnd);
     Optional<Record> getIndexRecord(String key, String indexName);
+    List<Record> getIndexRecords(List<String> keys, String indexName);
```

Comment:
Do we need this? This method only appears to be used below the `DataAccessLayer` interface.